src/storage/connectionStorage.ts

```diff
@@ -0,0 +1,314 @@
+      newStoreConnectionInfoWithSecrets as ConnectionInfoFromLegacyDS
+    );
+    const savedConnectionInfo = await this.saveConnection({
+      ...newStoreConnectionInfoWithSecrets,
+      connectionOptions: safeConnectionInfo.connectionOptions, // The connection info without secrets.
+    });
+    await this._storageController.setSecret(
+      savedConnectionInfo.id,
+      JSON.stringify(secrets)
+    );
+
+    return savedConnectionInfo;
+  }
+
+  async saveConnectionToStore(
+    storeConnectionInfo: StoreConnectionInfo
+  ): Promise<void> {
+    const variableName =
+      storeConnectionInfo.storageLocation === StorageLocation.GLOBAL
+        ? StorageVariables.GLOBAL_SAVED_CONNECTIONS
+        : StorageVariables.WORKSPACE_SAVED_CONNECTIONS;
+
+    // Get the current saved connections.
+    let savedConnections = this._storageController.get(
+      variableName,
+      storeConnectionInfo.storageLocation
+    );
+
+    if (!savedConnections) {
+      savedConnections = {};
+    }
+
+    // Add the new connection.
+    savedConnections[storeConnectionInfo.id] = storeConnectionInfo;
+
+    // Update the store.
+    return this._storageController.update(
+      variableName,
+      savedConnections,
+      storeConnectionInfo.storageLocation
+    );
+  }
+
+  async getStorageLocationFromPrompt() {
+    const storeOnWorkspace = 'Save the connection on this workspace';
+    const storeGlobally = 'Save the connection globally on vscode';
+    // Prompt the user where they want to save the new connection.
+    const chosenConnectionSavingLocation = await vscode.window.showQuickPick(
+      [
+        storeOnWorkspace,
+        storeGlobally,
+        "Don't save this connection (it will be lost when the session is closed)",
+      ],
+      {
+        placeHolder:
+          'Where would you like to save this new connection? (This message can be disabled in the extension settings.)',
+      }
+    );
+
+    if (chosenConnectionSavingLocation === storeOnWorkspace) {
+      return StorageLocation.WORKSPACE;
+    }
+
+    if (chosenConnectionSavingLocation === storeGlobally) {
+      return StorageLocation.GLOBAL;
+    }
+
+    return StorageLocation.NONE;
+  }
+
+  async loadConnections() {
+    const globalAndWorkspaceConnections = Object.values({
+      ...this._storageController.get(
+        StorageVariables.GLOBAL_SAVED_CONNECTIONS,
+        StorageLocation.GLOBAL
+      ),
+      ...this._storageController.get(
+        StorageVariables.WORKSPACE_SAVED_CONNECTIONS,
+        StorageLocation.WORKSPACE
+      ),
+    });
+
+    return (
+      await Promise.all(
+        globalAndWorkspaceConnections.map(async (connectionInfo) => {
+          const connectionInfoWithSecrets =
+            await this._getConnectionInfoWithSecrets(connectionInfo);
+          if (!connectionInfoWithSecrets) {
+            return;
+          }
+
+          return connectionInfoWithSecrets;
```

Comment:
No big deal, but just curious, how is this different from
```suggestion
          return await this._getConnectionInfoWithSecrets(connectionInfo);
```
?